src/Hydrodynamics/RiemannSolver.cpp

```diff
@@ -238,7 +238,7 @@ void ExactRiemannSolver<ndim>::ComputeStarRegion
     else if (2.0*fabs(pstar - pold)/(pstar + pold) < tolerance) break;
 
     // Check the star variables have not become NaNs
-    if (pstar != pstar || ustar != ustar) {
+    if (pstar != pstar) {
```

Comment:
What is this doing in this branch? What does it have to do with DeleteDeadParticles?